src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs

```diff
@@ -376,6 +376,35 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes, IHostEnvironment env,
                 ownModelFile: true, shapeDictionary: shapeDictionary, recursionLimit);
         }
 
+        /// <summary>
+        /// Create an OnnxModel from a <see cref="System.IO.Stream"/> of byte. Set execution to GPU if required.
+        /// Usually, an ONNX model is consumed by <see cref="OnnxModel"/> as a file.
+        /// With <see cref="CreateFromBytes(byte[], IHostEnvironment)"/> and
+        /// <see cref="CreateFromBytes(byte[], IHostEnvironment, int?, bool, IDictionary{string, int[]}, int)"/>,
+        /// it's possible to use in-memory model (type: byte[]) to create <see cref="OnnxModel"/>.
+        /// </summary>
+        /// <param name="modelBytes">Bytes of the serialized model.</param>
+        /// <param name="env">IHostEnvironment</param>
+        /// <param name="gpuDeviceId">GPU device ID to execute on. Null for CPU.</param>
+        /// <param name="fallbackToCpu">If true, resumes CPU execution quietly upon GPU error.</param>
+        /// <param name="shapeDictionary">User-provided shapes. If the key "myTensorName" is associated
+        /// with the value [1, 3, 5], the shape of "myTensorName" will be set to [1, 3, 5].
+        /// The shape loaded from <paramref name="modelBytes"/> would be overwritten.</param>
+        /// <param name="recursionLimit">Optional, specifies the Protobuf CodedInputStream recursion limit. Default value is 100.</param>
+        /// <returns>An <see cref="OnnxModel"/></returns>
+        public static OnnxModel CreateFromStream(Stream modelBytes, IHostEnvironment env, int? gpuDeviceId = null, bool fallbackToCpu = false,
+            IDictionary<string, int[]> shapeDictionary = null, int recursionLimit = 100)
+        {
+            var tempModelFile = Path.Combine(((IHostEnvironmentInternal)env).TempFilePath, Path.GetRandomFileName());
+            using (var fileStream = File.Create(tempModelFile))
+            {
+                modelBytes.Seek(0, SeekOrigin.Begin);
+                modelBytes.CopyTo(fileStream);
+            }
+            return new OnnxModel(tempModelFile, gpuDeviceId, fallbackToCpu,
```

Comment:
>OnnxModel [](http://example.com/codeflow?start=23&length=9)

I am wondering if we can have an OnnxModel constructor that can work directly with Streams? Or the native will not allow that? I am thinking if we can avoid writing to a temp file.

Reply:
Thats something I want to do as well, but this will at least unblock people and then I can come back and look into it. I am not sure if ONNX supports that or not (though they probably do).

Reply:
just a suggestion you may decide not to apply :-)

Instead of exposing `CreateFromStream`, would it make sense to expose a new constructor for `OnnxModel` which take the stream and make the implementation inside this constructor as the one provided here? This will avoid exposing extra method if we decided later to support streams in OnnxModel. This is minor point though.